tests/test_api_call.py

```diff
@@ -42,13 +48,134 @@ async def test_health_check(joint_fixture: JointFixture):  # noqa: F811
 
 
 @pytest.mark.asyncio
-async def test_api_call(monkeypatch, joint_fixture: JointFixture):  # noqa: F811
+async def test_api_calls(monkeypatch, joint_fixture: JointFixture):  # noqa: F811
     """Test functionality with incoming API call"""
+    file_secret = base64.b64encode(os.urandom(32)).decode("utf-8")
+    secret_id = "test_secret_id"
+    headers = {"Authorization": f"Bearer {joint_fixture.token}"}
+
+    encrypted_secret = EncryptedPayload(
+        payload=encrypt(data=file_secret, key=joint_fixture.keypair.public)
+    )
+
+    # test secret storage call
+    with monkeypatch.context() as patch:
+        # patch vault call with mock
+        patch.setattr(
+            "fis.adapters.outbound.vault.client.VaultAdapter.store_secret",
+            lambda self, secret: secret_id,
+        )
+        response = await joint_fixture.rest_client.post(
+            "/federated/ingest_secret",
+            json=encrypted_secret.dict(),
+            headers=headers,
+        )
+
+    assert response.status_code == 200
+    obtained_secret_id = json.loads(response.content)["secret_id"]
+    assert secret_id == obtained_secret_id
+
+    # test missing authorization
+    response = await joint_fixture.rest_client.post(
+        "/federated/ingest_metadata", json=encrypted_secret.dict()
+    )
+    assert response.status_code == 403
+
+    # test malformed payload
+    nonsense_payload = encrypted_secret.copy(update={"payload": "abcdefghijklmn"})
+    response = await joint_fixture.rest_client.post(
+        "/federated/ingest_metadata", json=nonsense_payload.dict(), headers=headers
+    )
+    assert response.status_code == 422
+
+    # test metadata ingest path
+    payload = UploadMetadata(
+        **joint_fixture.payload.dict(),
+        secret_id=secret_id,
+    )
+    encrypted_payload = EncryptedPayload(
+        payload=encrypt(
+            data=payload.json(),
+            key=joint_fixture.keypair.public,
+        )
+    )
+
     event_recorder = EventRecorder(
         kafka_servers=joint_fixture.kafka.config.kafka_servers,
         topic=joint_fixture.config.publisher_topic,
     )
 
+    with monkeypatch.context() as patch:
```

Comment:
Here you are not patching anything are you?

Reply:
Nope, missed that when adapting from the legacy version